plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script that automatically upgrades pieces of the Blockly API
+ *     that have been renamed by modifying developers' local files.
+ */
+'use strict';
+
+import {createSubCommand, extractRequiredInfo} from './command.js';
+import fetch from 'node-fetch';
+import {readFileSync, writeFileSync} from 'fs';
+import * as versionUtils from './versions.js';
+import JSON5 from 'json5';
+
+
+const DATABASE_URL = `https://raw.githubusercontent.com/google/blockly/develop/scripts/migration/renamings.json5`;
+
+export const rename = createSubCommand(
+    'rename', '>=5', 'Perform renamings for breaking changes')
+    .option(
+        '-i, --in-place [suffix]',
+        'do renamings in-place, optionally create backup files with the ' +
+        'given suffix. Otherwise output to stdout')
+    .action(async function() {
+      const {fromVersion, toVersion, fileNames} = extractRequiredInfo(this);
+
+      const renamer = new Renamer(await getDatabase(), fromVersion, toVersion);
+      fileNames.forEach((name) => {
+        const contents = readFileSync(name, 'utf8');
+        const newContents = renamer.rename(contents);
+        const i = this.opts().i;
+        if (i) {
+          if (typeof i == 'string') writeFileSync(name + i, contents);
+          writeFileSync(name, newContents);
+          process.stderr.write(`Migrated renamings in ${name}`);
+        } else {
+          process.stdout.write(newContents);
+        }
+      });
+    });
+
+/**
+ * Gets the database of renames.
+ * @return {!Promise<Object>} The database of renames as an object.
+ */
+export async function getDatabase() {
+  const response = await fetch(DATABASE_URL);
+  const body = await response.text();
+  return JSON5.parse(body);
+}
+
+/**
+ * Upgrades pieces of the Blockly API that have been renamed by modifying
+ * developers' local files.
+ * @param {!Object} database The database of renames.
+ * @param {string} currVersion The version to migrate from.
+ * @param {string} newVersion The version to migrate to.
+ * @param {!Array<string>} strings The strings to apply the renamings in.
+ * @return {!Array<string>} The strings with renamings applied.
+ * @deprecated Use Renamer instance instead.
```

Comment:
Why add a function that is already marked deprecated? i'm a little confused here

Reply:
Whoops, this was how we were structuring the API before, and it should be removed how that we added the classes. Thanks for the catch!